heracles/result.py

```diff
@@ -79,8 +79,13 @@ def __array_finalize__(self, obj: NDArray[Any] | None) -> None:
         self.upper = getattr(obj, "upper", None)
         self.weight = getattr(obj, "weight", None)
 
-    def __array_wrap__(self, context=None, return_scalar=False):
-        return super().__array_wrap__(context, return_scalar).view(np.ndarray)
+    def __array_wrap__(self, arr, context=None, return_scalar=False):
+        out = super().__array_wrap__(arr, context)
```

Comment:
Why not let the super `__array_wrap__` handle scalars?
```suggestion
        out = super().__array_wrap__(arr, context, return_scalar=return_scalar)
```

Reply:
Wouldn't work with NumPy 1 (no return_scalars there). And we need to target Python 3.9 and NumPy 1.

Reply:
Oh I see, the library supports NumPy v1 too.